### PR TITLE
[sonos] Add support for Sonos Roam

### LINKED ISSUE #10588: [sonos] Support for SONOS Roam
The new Sonos Roam is not supported properly 

I just added all the information I though I can provide initially. PLease hit me up, if you need more information.  

Result from http://192.168.X.XX:1400/xml/device_description.xml
`<root xmlns="urn:schemas-upnp-org:device-1-0">
<specVersion>
<major>1</major>
<minor>0</minor>
</specVersion>
<device>
<deviceType>urn:schemas-upnp-org:device:ZonePlayer:1</deviceType>
<friendlyName>192.168.X.XX - Sonos Roam</friendlyName>
<manufacturer>Sonos, Inc.</manufacturer>
<manufacturerURL>http://www.sonos.com</manufacturerURL>
<modelNumber>S27</modelNumber>
<modelDescription>Sonos Roam</modelDescription>
<modelName>Sonos Roam</modelName>
<modelURL>http://www.sonos.com/products/zoneplayers/S27</modelURL>
<softwareVersion>62.1-88070</softwareVersion>
<swGen>2</swGen>
<hardwareVersion>1.33.1.8-2.9</hardwareVersion>
<serialNum>XX-XX-XX-XX-XX-XX:A</serialNum>
<MACAddress>XX-XX-XX-XX-XX-XX</MACAddress>
<UDN>uuid:RINCON_XXXXXXXXXXXXX1400</UDN>
<iconList>
<icon>
<id>0</id>
<mimetype>image/png</mimetype>
<width>48</width>
<height>48</height>
<depth>24</depth>
<url>/img/icon-S27.png</url>
</icon>
</iconList>
<minCompatibleVersion>61.0-00000</minCompatibleVersion>
<legacyCompatibleVersion>58.0-00000</legacyCompatibleVersion>
<apiVersion>1.23.0-beta.1</apiVersion>
<minApiVersion>1.1.0</minApiVersion>
<displayVersion>13.0.5</displayVersion>
<extraVersion/>
<nsVersion>25</nsVersion>
<roomName>Sonos Roam</roomName>
<displayName>Roam</displayName>
<zoneType>29</zoneType>
<feature1>0x00000000</feature1>
<feature2>0x47c01312</feature2>
<feature3>0x0001cc8a</feature3>
<seriesid>A100</seriesid>
<variant>9</variant>
<internalSpeakerSize>5</internalSpeakerSize>
<bassExtension>0.000</bassExtension>
<satGainOffset>0.000</satGainOffset>
<memory>1024</memory>
<flash>1024</flash>
<flashRepartitioned>0</flashRepartitioned>
<ampOnTime>20</ampOnTime>
<retailMode>0</retailMode>
<SSLPort>1443</SSLPort>
<securehhSSLPort>1843</securehhSSLPort>
<serviceList>
<service>
<serviceType>urn:schemas-upnp-org:service:AlarmClock:1</serviceType>
<serviceId>urn:upnp-org:serviceId:AlarmClock</serviceId>
<controlURL>/AlarmClock/Control</controlURL>
<eventSubURL>/AlarmClock/Event</eventSubURL>
<SCPDURL>/xml/AlarmClock1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:MusicServices:1</serviceType>
<serviceId>urn:upnp-org:serviceId:MusicServices</serviceId>
<controlURL>/MusicServices/Control</controlURL>
<eventSubURL>/MusicServices/Event</eventSubURL>
<SCPDURL>/xml/MusicServices1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:AudioIn:1</serviceType>
<serviceId>urn:upnp-org:serviceId:AudioIn</serviceId>
<controlURL>/AudioIn/Control</controlURL>
<eventSubURL>/AudioIn/Event</eventSubURL>
<SCPDURL>/xml/AudioIn1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:DeviceProperties:1</serviceType>
<serviceId>urn:upnp-org:serviceId:DeviceProperties</serviceId>
<controlURL>/DeviceProperties/Control</controlURL>
<eventSubURL>/DeviceProperties/Event</eventSubURL>
<SCPDURL>/xml/DeviceProperties1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:SystemProperties:1</serviceType>
<serviceId>urn:upnp-org:serviceId:SystemProperties</serviceId>
<controlURL>/SystemProperties/Control</controlURL>
<eventSubURL>/SystemProperties/Event</eventSubURL>
<SCPDURL>/xml/SystemProperties1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:ZoneGroupTopology:1</serviceType>
<serviceId>urn:upnp-org:serviceId:ZoneGroupTopology</serviceId>
<controlURL>/ZoneGroupTopology/Control</controlURL>
<eventSubURL>/ZoneGroupTopology/Event</eventSubURL>
<SCPDURL>/xml/ZoneGroupTopology1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:GroupManagement:1</serviceType>
<serviceId>urn:upnp-org:serviceId:GroupManagement</serviceId>
<controlURL>/GroupManagement/Control</controlURL>
<eventSubURL>/GroupManagement/Event</eventSubURL>
<SCPDURL>/xml/GroupManagement1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-tencent-com:service:QPlay:1</serviceType>
<serviceId>urn:tencent-com:serviceId:QPlay</serviceId>
<controlURL>/QPlay/Control</controlURL>
<eventSubURL>/QPlay/Event</eventSubURL>
<SCPDURL>/xml/QPlay1.xml</SCPDURL>
</service>
</serviceList>
<deviceList>
<device>
<deviceType>urn:schemas-upnp-org:device:MediaServer:1</deviceType>
<friendlyName>192.168.X.XX - Sonos Roam Media Server</friendlyName>
<manufacturer>Sonos, Inc.</manufacturer>
<manufacturerURL>http://www.sonos.com</manufacturerURL>
<modelNumber>S27</modelNumber>
<modelDescription>Sonos Roam Media Server</modelDescription>
<modelName>Sonos Roam</modelName>
<modelURL>http://www.sonos.com/products/zoneplayers/S27</modelURL>
<UDN>uuid:RINCON_XXXXXXXXXXXXX1400_MS</UDN>
<serviceList>
<service>
<serviceType>urn:schemas-upnp-org:service:ContentDirectory:1</serviceType>
<serviceId>urn:upnp-org:serviceId:ContentDirectory</serviceId>
<controlURL>/MediaServer/ContentDirectory/Control</controlURL>
<eventSubURL>/MediaServer/ContentDirectory/Event</eventSubURL>
<SCPDURL>/xml/ContentDirectory1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:ConnectionManager:1</serviceType>
<serviceId>urn:upnp-org:serviceId:ConnectionManager</serviceId>
<controlURL>/MediaServer/ConnectionManager/Control</controlURL>
<eventSubURL>/MediaServer/ConnectionManager/Event</eventSubURL>
<SCPDURL>/xml/ConnectionManager1.xml</SCPDURL>
</service>
</serviceList>
</device>
<device>
<deviceType>urn:schemas-upnp-org:device:MediaRenderer:1</deviceType>
<friendlyName>Sonos Roam - Sonos Roam Media Renderer</friendlyName>
<manufacturer>Sonos, Inc.</manufacturer>
<manufacturerURL>http://www.sonos.com</manufacturerURL>
<modelNumber>S27</modelNumber>
<modelDescription>Sonos Roam Media Renderer</modelDescription>
<modelName>Sonos Roam</modelName>
<modelURL>http://www.sonos.com/products/zoneplayers/S27</modelURL>
<UDN>uuid:RINCON_XXXXXXXXXXXXX1400_MR</UDN>
<serviceList>
<service>
<serviceType>urn:schemas-upnp-org:service:RenderingControl:1</serviceType>
<serviceId>urn:upnp-org:serviceId:RenderingControl</serviceId>
<controlURL>/MediaRenderer/RenderingControl/Control</controlURL>
<eventSubURL>/MediaRenderer/RenderingControl/Event</eventSubURL>
<SCPDURL>/xml/RenderingControl1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:ConnectionManager:1</serviceType>
<serviceId>urn:upnp-org:serviceId:ConnectionManager</serviceId>
<controlURL>/MediaRenderer/ConnectionManager/Control</controlURL>
<eventSubURL>/MediaRenderer/ConnectionManager/Event</eventSubURL>
<SCPDURL>/xml/ConnectionManager1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:AVTransport:1</serviceType>
<serviceId>urn:upnp-org:serviceId:AVTransport</serviceId>
<controlURL>/MediaRenderer/AVTransport/Control</controlURL>
<eventSubURL>/MediaRenderer/AVTransport/Event</eventSubURL>
<SCPDURL>/xml/AVTransport1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-sonos-com:service:Queue:1</serviceType>
<serviceId>urn:sonos-com:serviceId:Queue</serviceId>
<controlURL>/MediaRenderer/Queue/Control</controlURL>
<eventSubURL>/MediaRenderer/Queue/Event</eventSubURL>
<SCPDURL>/xml/Queue1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:GroupRenderingControl:1</serviceType>
<serviceId>urn:upnp-org:serviceId:GroupRenderingControl</serviceId>
<controlURL>/MediaRenderer/GroupRenderingControl/Control</controlURL>
<eventSubURL>/MediaRenderer/GroupRenderingControl/Event</eventSubURL>
<SCPDURL>/xml/GroupRenderingControl1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:VirtualLineIn:1</serviceType>
<serviceId>urn:upnp-org:serviceId:VirtualLineIn</serviceId>
<controlURL>/MediaRenderer/VirtualLineIn/Control</controlURL>
<eventSubURL>/MediaRenderer/VirtualLineIn/Event</eventSubURL>
<SCPDURL>/xml/VirtualLineIn1.xml</SCPDURL>
</service>
</serviceList>
<X_Rhapsody-Extension xmlns="http://www.real.com/rhapsody/xmlns/upnp-1-0">
<deviceID>urn:rhapsody-real-com:device-id-1-0:sonos_1:RINCON_XXXXXXXXXXXXX1400</deviceID>
<deviceCapabilities>
<interactionPattern type="real-rhapsody-upnp-1-0"/>
</deviceCapabilities>
</X_Rhapsody-Extension>
<qq:X_QPlay_SoftwareCapability xmlns:qq="http://www.tencent.com">QPlay:2</qq:X_QPlay_SoftwareCapability>
<iconList>
<icon>
<mimetype>image/png</mimetype>
<width>48</width>
<height>48</height>
<depth>24</depth>
<url>/img/icon-S27.png</url>
</icon>
</iconList>
</device>
</deviceList>
</device>
</root>`

Result from http://192.168.X.XX:1400/xml/RenderingControl1.xml

`This XML file does not appear to have any style information associated with it. The document tree is shown below.
<scpd xmlns="urn:schemas-upnp-org:service-1-0">
<specVersion>
<major>1</major>
<minor>0</minor>
</specVersion>
<serviceStateTable>
<stateVariable sendEvents="yes">
<name>LastChange</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>Mute</name>
<dataType>boolean</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>Volume</name>
<dataType>ui2</dataType>
<allowedValueRange>
<minimum>0</minimum>
<maximum>100</maximum>
<step>1</step>
</allowedValueRange>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_LeftVolume</name>
<dataType>ui2</dataType>
<allowedValueRange>
<minimum>0</minimum>
<maximum>100</maximum>
<step>1</step>
</allowedValueRange>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_RightVolume</name>
<dataType>ui2</dataType>
<allowedValueRange>
<minimum>0</minimum>
<maximum>100</maximum>
<step>1</step>
</allowedValueRange>
</stateVariable>
<stateVariable sendEvents="no">
<name>VolumeDB</name>
<dataType>i2</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>Bass</name>
<dataType>i2</dataType>
<allowedValueRange>
<minimum>-10</minimum>
<maximum>10</maximum>
<step>1</step>
</allowedValueRange>
</stateVariable>
<stateVariable sendEvents="no">
<name>Treble</name>
<dataType>i2</dataType>
<allowedValueRange>
<minimum>-10</minimum>
<maximum>10</maximum>
<step>1</step>
</allowedValueRange>
</stateVariable>
<stateVariable sendEvents="no">
<name>EQValue</name>
<dataType>i2</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_EQType</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>Loudness</name>
<dataType>boolean</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>SupportsOutputFixed</name>
<dataType>boolean</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>OutputFixed</name>
<dataType>boolean</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>HeadphoneConnected</name>
<dataType>boolean</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_Channel</name>
<dataType>string</dataType>
<allowedValueList>
<allowedValue>Master</allowedValue>
<allowedValue>LF</allowedValue>
<allowedValue>RF</allowedValue>
</allowedValueList>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_MuteChannel</name>
<dataType>string</dataType>
<allowedValueList>
<allowedValue>Master</allowedValue>
<allowedValue>LF</allowedValue>
<allowedValue>RF</allowedValue>
<allowedValue>SpeakerOnly</allowedValue>
</allowedValueList>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_InstanceID</name>
<dataType>ui4</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_VolumeAdjustment</name>
<dataType>i4</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_RampType</name>
<dataType>string</dataType>
<allowedValueList>
<allowedValue>SLEEP_TIMER_RAMP_TYPE</allowedValue>
<allowedValue>ALARM_RAMP_TYPE</allowedValue>
<allowedValue>AUTOPLAY_RAMP_TYPE</allowedValue>
</allowedValueList>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_RampTimeSeconds</name>
<dataType>ui4</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_ResetVolumeAfter</name>
<dataType>boolean</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_ProgramURI</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>A_ARG_TYPE_ChannelMap</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>AudioDelay</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>AudioDelayLeftRear</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>AudioDelayRightRear</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>DialogLevel</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>SpeakerSize</name>
<dataType>ui4</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>SubCrossover</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>SubEnabled</name>
<dataType>boolean</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>SubGain</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>SubPolarity</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>SurroundLevel</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>MusicSurroundLevel</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>NightMode</name>
<dataType>boolean</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>SurroundEnabled</name>
<dataType>boolean</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>SurroundMode</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>PresetNameList</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>RoomCalibrationID</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>RoomCalibrationCoefficients</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>RoomCalibrationCalibrationMode</name>
<dataType>string</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>RoomCalibrationEnabled</name>
<dataType>boolean</dataType>
</stateVariable>
<stateVariable sendEvents="no">
<name>RoomCalibrationAvailable</name>
<dataType>boolean</dataType>
</stateVariable>
</serviceStateTable>
<actionList>
<action>
<name>GetMute</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_MuteChannel</relatedStateVariable>
</argument>
<argument>
<name>CurrentMute</name>
<direction>out</direction>
<relatedStateVariable>Mute</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetMute</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_MuteChannel</relatedStateVariable>
</argument>
<argument>
<name>DesiredMute</name>
<direction>in</direction>
<relatedStateVariable>Mute</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>ResetBasicEQ</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Bass</name>
<direction>out</direction>
<relatedStateVariable>Bass</relatedStateVariable>
</argument>
<argument>
<name>Treble</name>
<direction>out</direction>
<relatedStateVariable>Treble</relatedStateVariable>
</argument>
<argument>
<name>Loudness</name>
<direction>out</direction>
<relatedStateVariable>Loudness</relatedStateVariable>
</argument>
<argument>
<name>LeftVolume</name>
<direction>out</direction>
<relatedStateVariable>A_ARG_TYPE_LeftVolume</relatedStateVariable>
</argument>
<argument>
<name>RightVolume</name>
<direction>out</direction>
<relatedStateVariable>A_ARG_TYPE_RightVolume</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>ResetExtEQ</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>EQType</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_EQType</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>GetVolume</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_Channel</relatedStateVariable>
</argument>
<argument>
<name>CurrentVolume</name>
<direction>out</direction>
<relatedStateVariable>Volume</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetVolume</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_Channel</relatedStateVariable>
</argument>
<argument>
<name>DesiredVolume</name>
<direction>in</direction>
<relatedStateVariable>Volume</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetRelativeVolume</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_Channel</relatedStateVariable>
</argument>
<argument>
<name>Adjustment</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_VolumeAdjustment</relatedStateVariable>
</argument>
<argument>
<name>NewVolume</name>
<direction>out</direction>
<relatedStateVariable>Volume</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>GetVolumeDB</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_Channel</relatedStateVariable>
</argument>
<argument>
<name>CurrentVolume</name>
<direction>out</direction>
<relatedStateVariable>VolumeDB</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetVolumeDB</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_Channel</relatedStateVariable>
</argument>
<argument>
<name>DesiredVolume</name>
<direction>in</direction>
<relatedStateVariable>VolumeDB</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>GetVolumeDBRange</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_Channel</relatedStateVariable>
</argument>
<argument>
<name>MinValue</name>
<direction>out</direction>
<relatedStateVariable>VolumeDB</relatedStateVariable>
</argument>
<argument>
<name>MaxValue</name>
<direction>out</direction>
<relatedStateVariable>VolumeDB</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>GetBass</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>CurrentBass</name>
<direction>out</direction>
<relatedStateVariable>Bass</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetBass</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>DesiredBass</name>
<direction>in</direction>
<relatedStateVariable>Bass</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>GetTreble</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>CurrentTreble</name>
<direction>out</direction>
<relatedStateVariable>Treble</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetTreble</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>DesiredTreble</name>
<direction>in</direction>
<relatedStateVariable>Treble</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>GetEQ</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>EQType</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_EQType</relatedStateVariable>
</argument>
<argument>
<name>CurrentValue</name>
<direction>out</direction>
<relatedStateVariable>EQValue</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetEQ</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>EQType</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_EQType</relatedStateVariable>
</argument>
<argument>
<name>DesiredValue</name>
<direction>in</direction>
<relatedStateVariable>EQValue</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>GetLoudness</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_Channel</relatedStateVariable>
</argument>
<argument>
<name>CurrentLoudness</name>
<direction>out</direction>
<relatedStateVariable>Loudness</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetLoudness</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_Channel</relatedStateVariable>
</argument>
<argument>
<name>DesiredLoudness</name>
<direction>in</direction>
<relatedStateVariable>Loudness</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>GetSupportsOutputFixed</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>CurrentSupportsFixed</name>
<direction>out</direction>
<relatedStateVariable>SupportsOutputFixed</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>GetOutputFixed</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>CurrentFixed</name>
<direction>out</direction>
<relatedStateVariable>OutputFixed</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetOutputFixed</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>DesiredFixed</name>
<direction>in</direction>
<relatedStateVariable>OutputFixed</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>GetHeadphoneConnected</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>CurrentHeadphoneConnected</name>
<direction>out</direction>
<relatedStateVariable>HeadphoneConnected</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>RampToVolume</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_Channel</relatedStateVariable>
</argument>
<argument>
<name>RampType</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_RampType</relatedStateVariable>
</argument>
<argument>
<name>DesiredVolume</name>
<direction>in</direction>
<relatedStateVariable>Volume</relatedStateVariable>
</argument>
<argument>
<name>ResetVolumeAfter</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_ResetVolumeAfter</relatedStateVariable>
</argument>
<argument>
<name>ProgramURI</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_ProgramURI</relatedStateVariable>
</argument>
<argument>
<name>RampTime</name>
<direction>out</direction>
<relatedStateVariable>A_ARG_TYPE_RampTimeSeconds</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>RestoreVolumePriorToRamp</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>Channel</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_Channel</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetChannelMap</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>ChannelMap</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_ChannelMap</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetRoomCalibrationX</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>CalibrationID</name>
<direction>in</direction>
<relatedStateVariable>RoomCalibrationID</relatedStateVariable>
</argument>
<argument>
<name>Coefficients</name>
<direction>in</direction>
<relatedStateVariable>RoomCalibrationCoefficients</relatedStateVariable>
</argument>
<argument>
<name>CalibrationMode</name>
<direction>in</direction>
<relatedStateVariable>RoomCalibrationCalibrationMode</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>GetRoomCalibrationStatus</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>RoomCalibrationEnabled</name>
<direction>out</direction>
<relatedStateVariable>RoomCalibrationEnabled</relatedStateVariable>
</argument>
<argument>
<name>RoomCalibrationAvailable</name>
<direction>out</direction>
<relatedStateVariable>RoomCalibrationAvailable</relatedStateVariable>
</argument>
</argumentList>
</action>
<action>
<name>SetRoomCalibrationStatus</name>
<argumentList>
<argument>
<name>InstanceID</name>
<direction>in</direction>
<relatedStateVariable>A_ARG_TYPE_InstanceID</relatedStateVariable>
</argument>
<argument>
<name>RoomCalibrationEnabled</name>
<direction>in</direction>
<relatedStateVariable>RoomCalibrationEnabled</relatedStateVariable>
</argument>
</argumentList>
</action>
</actionList>
</scpd>`

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #10588

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by martinvw** [comment]
LGTM, should we wait for @kgoderis's review given he is the official code owner?

**Comment by lolodomo** [comment]
@kgoderis was not really involved in the binding maintenance these last years. He is of course welcome to review but I would recommend to not wait for him too much.

**Review by kaikreuzer — APPROVED** [review]
I agree with @lolodomo, PR looks good.

**Pull request merged**
